src/mqt/yaqs/core/data_structures/hamiltonian_schedule.py

```
@@ -0,0 +1,147 @@
+    interval_specs: list[HamiltonianInterval] = []
+    validated: set[tuple[object, ...]] = set()
+
+    for start_raw, end_raw in zip(sim_params.times[:-1], sim_params.times[1:], strict=True):
+        start = float(start_raw)
+        end = float(end_raw)
+        duration = end - start
+        substep_duration = duration / sim_params.tdvp_sweeps
+        substeps: list[HamiltonianSubstep] = []
+        for sweep_index in range(sim_params.tdvp_sweeps):
+            midpoint = start + (sweep_index + 0.5) * substep_duration
+            parameters = hamiltonian._parameters_at(midpoint)  # ruff: ignore[private-member-access]
```

Ruff (0.16.1)

[warning] Invalid rule code in suppression: private-member-access

Enable `lint.preview` to use rule names

(RUF102)